BetterMenus/better-menus.md

```diff
@@ -0,0 +1,192 @@
+                    MenuRef.Update();
+                    
+                    Log("Name of above button changed"); //add a log to know my code worked
+                }),
+        });
+```
+This code produces the following modmenu
+[![Property Update Example](../Images/BetterMenusPropertyUpdateExample.jpg)](https://youtu.be/IlXgQSa3zTs)
+> Notice: Since I'm changing the name, I manually specified Id so there is no confusion.  
+
+> Note: The compiler might get an ambiguous reference between UnityEngine.UI.MenuButton and Satchel.BetterMenus.MenuButton. Best to do `using Satchel.BetterMenus` on the top to avoid this error
+### 3. The MenuRow Element
+The Menu Row element allows the modmenu to have multiple elements in a single horizontal line. Mostly used if you have multiple buttons and you dont want to waste vertical space on all of them
+Example
+```cs
+MenuRef = new Menu("Example Mod Menu", new Element[]
+        {
+            //create a menu row with 2 menu buttons. 
+            new MenuRow(new List<Element>() //first parameter is a list of elements you want to add to the row
+                {
+                    new MenuButton("ButtonL", "Left Button", (_) => Log("Left Button Pressed")),
+                    new MenuButton("ButtonR", "Right Button", (_) => Log("Right Button Pressed")),
+                }, "MyMenuButtonRow"), //second parameter is the id
+        });
+```
+![MenuRow Example](../Images/BetterMenusMenuRowExample.jpg)
+### 4. Hide/Showing Elements
+This feature is very useful for showing only relevant elements to the user based on the current selected options. Using this system is very similar to [updating properties](#2-updating-properties-of-elements).
+Every Element has a property `isVisible` that controls whether not not its placed on next Menu Update. An Example will make this more clear.
+
+Example:
+```cs
+ enum Modes
+{
+    Mode1,
+    Mode2,
+}
+
+Modes chosenMode = Modes.Mode1;
+MenuRef = new Menu("Example Mod Menu", new Element[]
+{
+    new HorizontalOption("Choose Mode", 
+        "Choose from the following modes",
+        Enum.GetNames(typeof(Modes)), //get an string array of all values in Modes enum
+        (index) => //the function that will be run when a new option is set
+        { 
+            chosenMode = (Modes)index;
+            switch (chosenMode)
+            {
+                case Modes.Mode1:
+                    MenuRef.Find("Mode1Key").Show(); //show the element
+                    MenuRef.Find("Mode2Button").Hide(); //hide the element
+                    break;
+                case Modes.Mode2:
+                    //Another option is to find the buttons and update its visibility property and call Update function.
+                    MenuRef.Find("Mode1Key").isVisible = chosenMode == Modes.Mode1; 
+                    MenuRef.Find("Mode2Button").isVisible = chosenMode == Modes.Mode2; 
```

Comment:
caching Elements that are found should be a part of the docs, probably should be the standard way in examples.

Reply:
But the Find is only being used twice? I don't see the problem

Reply:
Caching elements on top would break the readability of this example

Reply:
Note: It shows outdated but i havent addressed your issue for the reasons I gave